packages/vue-components/src/Navbar.vue

```diff
@@ -125,7 +125,7 @@ export default {
       if (uParts.length <= hParts.length) {
         return false;
       }
-      for (let i = 0; i < hParts.length; i += 1) {
+      for (let i = 0; i < hParts.length - 1; i += 1) {
```

Comment:
The sibling one is already functioning correctly if I'm not mistaken;

E.g. (since after `normalizeUrl` all urls end with `.html`)
`xx/yy/foo.html`
`xx/yy/bar.html`
we should check both `xx` and `yy`

In contrast the child check isn't without this change (try #1698's new subpages for an example):
E.g.
`xx/yy/foo.html`
`xx/yy/zz/bar.html`
we should check both `xx` and `yy`, but not `foo.html`

I added a new commit 5b93fd3 as well to remove the small code duplication between these 2 functions since the checks are identical.

Reply:
Ahh ok I was thinking along the lines of having `/index.html` appended to all normalized urls earlier.

But I believe there is still a case of having:
`xx/yy/foo/index.html`
`xx/yy/bar/index.html`
we would need to stop at `-2` for this

Reply:
> we would need to stop at -2 for this

👀 
Is this for `isSibling`?

Shouldn't it be up till `-1`, since in this case the two links are not siblings because `foo` !== `bar`

Reply:
Oh right XD Ignore the previous comment 😅😅